### PR TITLE
Search functionality implementation

### PR DESCRIPTION
**Pull request by Iyanuoluwa-abolade**
- Implemented a search that allows users to search for profiles by name
- Removed the rank present in favorites

https://pxl.cl/5m12N (link to video)
Image 8-1-24 at 11 33 PM
Image 8-1-24 at 4 40 PM (1)

### CODE REVIEW

File: frontend/my-adopthive-app/src/AdopterFavourites/AdopterFavourites.jsx [hunk 1]
```diff
@@ -84,21 +54,10 @@ const AdopterFavourites = () => {
                             <p>Dreams: {fav.adoptee.dreams}</p>
                             <p>City: {fav.adoptee.city}</p>
                             <p>Country: {fav.adoptee.country}</p>
-                            <div className="rank-container">
```
Comment (vandanasub) [inline frontend/my-adopthive-app/src/AdopterFavourites/AdopterFavourites.jsx]: thanks for cleaning this up!

File: backend/data.json [hunk 2]
```diff
@@ -0,0 +1,228 @@
+{
```
Comment (vandanasub) [inline backend/data.json:1]: is this different from what you added in PR8? Seems like there's some overlap in the commits, could you clarify?

**Comment by vandanasub** [comment]
Search feature looks great! Thanks for proactively thinking of this, it's a great addition to the app.

File: backend/data.json (hunk 2 above)
Comment (Iyanuoluwa-abolade) [reply backend/data.json:1]: No, it is the same with the one I added in PR 8

**Pull request merged**